ext/standard/string.c

```diff
@@ -1264,16 +1264,16 @@ PHPAPI void php_implode(const zend_string *glue, zval *pieces, zval *return_valu
    Joins array elements placing glue string between items and return one string */
 PHP_FUNCTION(implode)
 {
-	zval *arg1, *arg2 = NULL, *pieces;
+	zval *arg1, *pieces = NULL;
 	zend_string *glue, *tmp_glue;
 
 	ZEND_PARSE_PARAMETERS_START(1, 2)
 		Z_PARAM_ZVAL(arg1)
```

Comment:
I'm fairly sure this should just be changed to Z_PARAM_STR instead of throwing a special cased TypeError (it accepts a zend_string)

Also, I see no reason for it to continue to be optional. The whole block for `if (pieces == NULL) {` should also no longer be necessary - It will always be non-null after taking out Z_PARAM_OPTIONAL.

```patch
 	ZEND_PARSE_PARAMETERS_START(1, 2)
  		Z_PARAM_ZVAL(arg1)
-		Z_PARAM_OPTIONAL
-		Z_PARAM_ZVAL(arg2)
+		Z_PARAM_STRING(glue)
 		Z_PARAM_ARRAY(pieces)
 	ZEND_PARSE_PARAMETERS_END();
```

I don't see any other issues - there should be a note in UPGRADING "Backward Incompatible Changes", but I've seen that get added after PRs

Reply:
implode() has a single-parameter version where only the array is provided, which is not deprecated.